File: flycloud-ui/flycloud-admin-ui/src/api/mall/promotion/coupon/couponTemplate.ts
import request from '@/config/axios'

const MALL_BASE_URL = import.meta.env.VITE_MALL_SERVER

export interface CouponTemplateVO {
  id: number
  name: string
  status: number
  totalCount: number
  takeLimitCount: number
  takeType: number
  usePrice: number
  productScope: number
  productScopeValues: number[]
  validityType: number
  validStartTime: Date
  validEndTime: Date
  fixedStartTerm: number
  fixedEndTerm: number
  discountType: number
  discountPercent: number
  discountPrice: number
  discountLimitPrice: number
  takeCount: number
  useCount: number
}

// 创建优惠劵模板
export function createCouponTemplate(data: CouponTemplateVO) {
  return request.post({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/create`,
    data: data
  })
}

// 更新优惠劵模板
export function updateCouponTemplate(data: CouponTemplateVO) {
  return request.put({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/update`,
    data: data
  })
}

// 更新优惠劵模板的状态
export function updateCouponTemplateStatus(id: number, status: [0, 1]) {
  const data = {
    id,
    status
  }
  return request.put({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/update-status`,
    data: data
  })
}

// 删除优惠劵模板
export function deleteCouponTemplate(id: number) {
  return request.delete({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/delete?id=` + id
  })
}

// 获得优惠劵模板
export function getCouponTemplate(id: number) {
  return request.get({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/get?id=` + id
  })
}

// 获得优惠劵模板分页
export function getCouponTemplatePage(params: PageParam) {
  return request.get({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/page`,
    params: params
  })
}

// 获得优惠劵模板分页
export function getCouponTemplateList(ids: number[]): Promise<CouponTemplateVO[]> {
  return request.get({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/list?ids=${ids}`
  })
}

// 导出优惠劵模板 Excel
export function exportCouponTemplateExcel(params: PageParam) {
  return request.get({
    url: `/${MALL_BASE_URL}/promotion/coupon-template/export-excel`,
    params: params,
    responseType: 'blob'
  })
}

File: flycloud-api/flycloud-mall-api/src/main/java/com/fly/mall/api/promotion/domain/CouponTemplate.java
package com.fly.mall.api.promotion.domain;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.fly.common.domain.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 优惠券模板 表。
 *
 * @author lxs
 * @date 2026-07-02
 */
@TableName(value = "promotion_coupon_template", autoResultMap = true)
@Data
@EqualsAndHashCode(callSuper = true)
@Schema(name = "优惠券模板对象", description = "优惠券模板表")
public class CouponTemplate extends BaseEntity {

    @TableId
    private Long id;

    private String name;

    private String description;

    private Integer status;

    private Integer totalCount;

    private Integer takeLimitCount;

    private Integer takeType;

    private Integer usePrice;

    private Integer productScope;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<Long> productScopeValues;

    private Integer validityType;

    private LocalDateTime validStartTime;

    private LocalDateTime validEndTime;

    private Integer fixedStartTerm;

    private Integer fixedEndTerm;

    private Integer discountType;

    private Integer discountPercent;

    private Integer discountPrice;

    private Integer discountLimitPrice;

    private Integer takeCount;

    private Integer useCount;

}
